primap2/csg/_models.py

```diff
@@ -0,0 +1,168 @@
+"""Models for the composite source generator."""
+
+import typing
+from collections.abc import Hashable
+
+import numpy as np
+import xarray as xr
+from attr import define
+
+
+@define
+class PriorityDefinition:
+    """
+    Defines source priorities for composing a full dataset or a single timeseries.
+
+    Attributes
+    ----------
+    selection_dimensions
+        List of dimensions from which source timeseries are selected. These are the
+        priority dimensions, and each priority has to specify all selection dimensions.
+    priorities
+        List of priority selections. Higher priority selections come first. Each
+        selection consists of a dict which maps dimension names to values. Each
+        selection has to specify all selection_dimensions, but may specify additional
+        dimensions (fixed dimensions) to limit the selection to specific cases.
+        Examples:
+        [{"area (ISO3)": "COL", "source": "A"}, {"source": "B"}]
+        would select source "A" for Columbia, but source "B" for all other countries.
```

Comment:
Yes, that's correct, it will select source "A" as first priority for Columbia, and source "B" as second priority, but it would select only source "B" for all other countries. I'll update the description.